feat(replays): Show seconds in the tooltip when hovering Replay Details header

ryan953 opened the pull request:
Here's the tooltip with seconds displayed now:

<img width="390" alt="Screen Shot 2022-09-26 at 11 22 44 AM" src="https://user-images.githubusercontent.com/187460/192353461-ecc77183-12a8-4430-ade9-8162692f81c9.png">

Because the tooltip has a max-width, the content wraps which doesn't look the best. For that reason I made it the new format an option, **not the default**.


For exmple, if we changed the default then the Activity list would also start to wrap, see before & after:
| Current Tooltip | Display seconds in tooltip (not implemented) |
| ------------- | ------------- |
| <img width="390" alt="Screen Shot 2022-09-26 at 11 29 37 AM" src="https://user-images.githubusercontent.com/187460/192353760-94bdad9e-1aec-4117-9510-01553fa7b3e6.png">  | <img width="318" alt="Screen Shot 2022-09-26 at 11 25 33 AM" src="https://user-images.githubusercontent.com/187460/192353764-c14301b2-6b43-4f2a-8ce3-9551cfad8989.png">  |

ryan953 committed:
feat(replays): Show seconds in the tooltip when hovering Replay Details header

ryan953 commented on static/app/components/timeSince.tsx:
```diff
@@ -82,7 +84,10 @@ function TimeSince({
+  const tooltipFormat = tooltipShowSeconds
+    ? 'MMMM D, YYYY h:mm:ss A z'
+    : 'MMMM D, YYYY h:mm A z';
+  const format = options?.clock24Hours ? 'MMMM D, YYYY HH:mm z' : tooltipFormat;
```
https://momentjs.com/docs/#/displaying/

evanpurkhiser commented:
Should we maybe use short month name so it doesn’t wrap?

billyvg reviewed (commented):
I almost wonder if we should accept arbitrary formatting here. Maybe not since our date formatting should be as consistent as possible, and Replays is a one-off that would benefit from displaying seconds.

The wrapping is not great -- since this was an issue I brought up randomly, let's wait for post DEX to merge. (I think ideally we would increase the max-width?).

ryan953 commented:
I was thinking that increasing the max-width is the easiest. It seems like it's a random number at 225px right now. 
But increasing it ~whenever isn't the best strategy.

shruthilayaj committed:
fix(discover): Add a loading/reloading state to metrics baseline (#39306)

This is to handle visual jank a little better when switching
projects. Also update beta badge.

emmatyping committed:
fix(admin): Fix typo in admin user creation form (#38418)


This fixes the error reported in https://github.com/getsentry/sentry/issues/38303, which appears to be due to a typo in the django module name.

Commit:
fix(hovercard): Add overflow to code segment in FeatureDisabled (#39177)

Before:
<img width="670" alt="image"
src="https://user-images.githubusercontent.com/9372512/191872141-5725c360-53ea-4577-8333-305e1cf7e5d8.png">

After:
<img width="508" alt="image"
src="https://user-images.githubusercontent.com/9372512/191872125-d617047d-e1bd-4e54-a8c5-e4ee447032ea.png">

evanpurkhiser committed:
test(acceptance): Set an extra garbage cookie (#39273)

wedamija committed:
fix(tests): Use `RedisSnubaTSDB` by default in all tests (#39297)

`RedisSnubaTSDB` has been the default in productions. To make our tests
reflect production we should use it there as well.

Removed most uses of `tsdb.incr` from the tests. The only ones left are
places that are actually still using `tsdb.incr`.

evanpurkhiser committed:
ref(js): Use absolute import for persistedStoreProvider (#39308)

asottile-sentry committed:
ref: disable flaky test_count_transaction_with_valid_condition (#39295)

https://getsentry.atlassian.net/browse/TET-423

wedamija committed:
refs(tests): Add missing __init__.py files (#39317)

dashed committed:
feat(csrf): Add analytics for CSRF view failures (#39311)

scttcper committed:
fix(ecosystem): Do not say "synced" on ownership rules (#39319)

scttcper committed:
feat(ecosystem): Allow org members to add code mappings (#39172)

vuluongj20 committed:
ref(tabs): Remove unnecessary orientation attributes (#39315)

`orientation` is a styling prop and should not appear as an attribute in
the DOM. Only the tab list element needs an `aria-orientation`
attribute.

evanpurkhiser committed:
feat(js): Add `transformData` to Form Models (#39313)

evanpurkhiser committed:
ref(js): Move AttachmentUrl component to components (#39325)

narsaynorath committed:
fix(discover): Flicker due to fallback to stale savedQuery (#39303)

When the user clicks "Save Changes", we redirect them to a short URL. At
this point, savedQuery from the AsyncComponent becomes stale (because
we've updated but haven't fetched the new query). The Results component
then tries to use this stale query to build an eventView, but since
we've redirected to a short URL the changes aren't there to override the
values properly.

Instead, when the update happens successfully, save the response so we
derive the proper eventView.

narsaynorath committed:
feat(discover): Add discover homepage query endpoint (#39153)

Adds `/api/0/:org-id/discover/homepage/` as a private endpoint. This
endpoint supports GET, PUT, and DELETE

gggritso committed:
fix(perf-issues): Tag metrics with integrations (#39309)

Tag the `"performance.performance_issue.detected"` metric with four new
boolean flags corresponding to common Python SDK integrations. Adds tags
like `"integration_django"`, `"integration_flask"`, etc.

scttcper committed:
fix(ecosystem): Cleanup unused codeowners CTA (#39294)

armenzg committed:
fix(gitlab): Fix AppPlatformEvents hitting Gitlab webhook endpoint (#39333)

It seems that AppPlatformEvents hits the Gitlab webhook [1] and it does
not contain the Gitlab headers. This regression was introduced via
#39129.

Fixes
[SENTRY-W3G](https://sentry.io/organizations/sentry/issues/3624081212/?project=1)

[1]
https://github.com/getsentry/sentry/blob/07362b7d8eb6b03acc21aafb91e58a1865c00da3/src/sentry/api/serializers/models/app_platform_event.py#L46-L55

gggritso committed:
fix(perf-issues): Tag transaction save event with SDK name (#39334)

Allows for finding transaction by SDK for debugging.

0Calories committed:
fix(perf-issues): Fix spans being too far out of view horizontally (#39162)

wmak committed:
feat(discover): Add interval badge (#39320)

- This adds a FeatureBadge for the IntervalSelector feature

### Preview
<img width="1523" alt="image"
src="https://user-images.githubusercontent.com/4205004/192398560-705c49dd-87e3-45e5-af3b-a5c36f959140.png">

ryan953 committed:
fix(replays): Make sure the url walker scrolls when the list is large (#39279)

Checked that scrolling on both the Replay Detail Page and List pages
work, and for smaller lists everything looks good too.

<img width="341" alt="Screen Shot 2022-09-25 at 11 36 18 AM"
src="https://user-images.githubusercontent.com/187460/192159821-001a1523-b151-4dad-b630-7c9ed4d60c82.png">
<img width="391" alt="Screen Shot 2022-09-25 at 11 37 28 AM"
src="https://user-images.githubusercontent.com/187460/192159822-f332c698-2e63-4280-ad92-3633be245754.png">

wedamija committed:
fix(tests): Fix flaky stats tests (#39345)

These tests all fail if run in the first minute of the hour. Tested this
with `@freeze_time(before_now(days=1).replace(minute=0))`. For tests
that fail at that time I've hardcoded them to run at a specific minute
of the hour.

roggenkemper committed:
feat(growth): export iconarrow for sandbox (#39310)

this pr exports the arrow icon for use in the sandbox

lynnagara committed:
feat: Add arroyo compatible synchronized consumer (#39340)

This brings over the synchronized consumer code that was previously part of arroyo. It was removed in https://github.com/getsentry/arroyo/pull/81 since it was no longer being used by Snuba and should not live in the shared library.

This implementation differs from the current synchronized consumer code however it is functionally the same. This code has been fairly extensively tested (as it was previously used with Snuba subscriptions) It is currently not used but will be used when converting the post process forwarder to use Arroyo.

gggritso committed:
fix(perf-issues): Fix errors with integration tags on save event (#39347)

- Fall back to empty integration list if the SDKs returns `None`
- Lowercase integration names

wmak committed:
feat(discover): Add received to the backend (#38414)

- this adds the received field to discover's backend
- pt2: https://github.com/getsentry/sentry/pull/38415

RyanSkonnord committed:
fix(hybrid): Tag some models as available on all silos (#38785)

Commit:
ref(monitors): Replace IssueList with GroupList (#39323)

Monitors was the only page using the `IssueList` component which is very
redundant with the existing `GroupList` component, so we should unify
and then get rid of `IssueList`
(https://github.com/getsentry/sentry/pull/39324)

Before:
<img width="1184" alt="image"
src="https://user-images.githubusercontent.com/9372512/192406567-871c920d-80e1-44d3-9521-34202aad7f39.png">

After:
<img width="1188" alt="image"
src="https://user-images.githubusercontent.com/9372512/192406532-b7040bb2-9f5d-4399-a118-a63f864b8d9b.png">

wedamija committed:
refs(tests): Remove `USE_SNUBA` env variable from test configs. (#39316)

This is unused since we merged
https://github.com/getsentry/sentry/pull/39297. So both the backend and
snuba tests are running on almost the same config now. The main
difference is that snuba tests run kafka, while backend don't.

We could probably look into unifying these jobs in the near future.

evanpurkhiser committed:
ref(js): Remove UNSAFE_componentWillReceiveProps from AvatarChooser (#39215)

evanpurkhiser committed:
ref(js): Remove UNSAFE_componentWillReceiveProps from group/pluginActions (#39217)

vuluongj20 committed:
fix(issueDetails): Highlight Tags tab in Tag Details page (#39348)

**Before:** we're in the Tag Details page, but the "Details" tab is
highlighted
<img width="833" alt="Screen Shot 2022-09-27 at 10 49 16 AM"
src="https://user-images.githubusercontent.com/44172267/192599880-60c859f9-49d5-422e-adf6-7223ea2b2739.png">

**After:** the "Tags" tab is correctly highlighted
<img width="833" alt="Screen Shot 2022-09-27 at 10 49 01 AM"
src="https://user-images.githubusercontent.com/44172267/192599832-9e7134eb-53ed-4e6b-8342-28d62fcd6109.png">

malwilley committed:
chore(ui): Disable direct autoimport of testing-library libs (#39346)

Things like `fireEvent` and some other imports would appear at the top
and we should always import from `sentry-test` instead.

RyanSkonnord committed:
feat(hybrid): Introduce region function contracts (#39253)

Set up a type system for region objects and the set of all regions, with
placeholders for backend operations not implemented yet.

malwilley committed:
test(ui): Convert DropdownLink tests to RTL and TS (#39322)

getsentry-bot committed:
Revert "feat(hybrid): Introduce region function contracts (#39253)"

This reverts commit a7d5d842763726afd7d57ac5f952425ad8dc825b.

Co-authored-by: anthony.sottile via Slack <anthony.sottile@sentry.io>

phacops committed:
feat(profiling): Process sample format (#39106)

evanpurkhiser committed:
ref(js): Improve feature badge spacing (#39352)

evanpurkhiser committed:
ref(js): Retain constructed form Model options (#39351)

RyanSkonnord committed:
feat(hybrid): Introduce region function contracts (#39356)

Set up a type system for region objects and the set of all regions, with
placeholders for backend operations not implemented yet.

Remaking changes from #39253, which was reverted due to an error
now fixed by getsentry/getsentry#8407

Commit:
feat(commit-context): Add logging to debug API fetches (#39357)

## Objective:
Add logging to determine why the API is not returning matches.

phacops committed:
fix(profiling): Fix typo in device classifcation type (#39360)

growupanand committed:
fix(avatarChooser): Handle submit without selecting file #38190 (#38191)

Co-authored-by: Utkarsh Anand <utkarsh.anand@greendeck.co>
Co-authored-by: Evan Purkhiser <evanpurkhiser@gmail.com>

barkbarkimashark committed:
fix(releases): fix release version not validating none value (#39358)

Missed a spot when validating the release version value.

Fixes SENTRY-VMK

scttcper committed:
feat(workflow): Set project ownership fallthrough default false (#39305)

robinrendle committed:
fix(copy): Copy and design of Issue Owners page (#39318)

- Copy fixes to the modal to make it easier to read
- Copy fixes to settings page

robinrendle committed:
fix(copy): event -> event id on issue details page (#39365)

We should rename `Event` to `Event ID` because everything beneath this
text is information related to the Event

## Before

<img width="850" alt="CleanShot 2022-09-27 at 13 19 04@2x"
src="https://user-images.githubusercontent.com/1900676/192627747-8e053b78-b30f-4638-873a-59dff7de9f6e.png">

## After

<img width="840" alt="CleanShot 2022-09-27 at 13 18 42@2x"
src="https://user-images.githubusercontent.com/1900676/192627787-6140f1ed-96db-433f-b42e-fc6fc180dbb0.png">

vuluongj20 committed:
fix(issueDetails): Make tab panels span full height (#39314)

**Before:**
![Screen Shot 2022-09-26 at 15 52
09](https://user-images.githubusercontent.com/44172267/192394527-f9c71cee-09de-4a9e-b194-e18b5373dfaa.png)

**After:**
![Screen Shot 2022-09-26 at 15 52
00](https://user-images.githubusercontent.com/44172267/192394525-98ac1834-3f28-44ae-bd0e-4028a65d9ea1.png)

wmak committed:
fix(discover): Ensure dropdown indexes are unique (#39361)

- This fixes a bug where results.length would be reduced by 2 since we
remove numbers out of the min&max bounds, so results.length + 1 would be
a used index for makeItem

wmak committed:
chore(querybuilder): Consolidate filter aliases (#39342)

- Moving all the shared filter aliases to the filter_aliases.py file so
that they're together

phacops committed:
fix(profiling): Apply results of symbolication for cocoa profiles (#39367)

corps committed:
feat(hc): Publish UserIP logs to kafka from REGION silos (#38988)

Enable feature flag backed ability to publish UserIP entries to a kafka producer associated with control silos, which consume and persist them.

asottile-sentry committed:
ref: exhaust streaming response in test_charcuterie (#39372)

this _might_ be the leaking file, haven't actually dug into it but
decided to grep around first

asottile-sentry committed:
ref: close the debug-files streaming response (#39373)

<!-- Describe your PR here. -->

malwilley committed:
fix(ui): Fix overflowing release details filters on mobile (#39377)

scttcper committed:
fix(issues): Add padding to stacktrace key/values (#39380)

evanpurkhiser committed:
ref(js): Remove `memberListStore.getByEmail` (#39376)

evanpurkhiser committed:
ref(js): Replace PublishRequestFormModel w/ transformData opt (#39326)

evanpurkhiser committed:
ref(js): Add missing type to tabs util (#39384)

evanpurkhiser committed:
ref(js): Replace SentryFunctionFormModel w/ transformData opt (#39327)

barkbarkimashark committed:
feat(perf-issues): execute search queries for various GroupCategories in parallel (#39274)

This PR adds the ability to search multiple datasources in parallel if
the search criteria requires it.

To achieve the parallelism, we leverage the already existing method
`bulk_raw_query` which already submits queries in a threadpool. We
simply needed to:
1. determine if we need to query multiple data sources
2. introduce intermediate functions that extracted out the query
parameters (instead of directly submitting and returning the data as
seen in `aliased_query` and `raw_query`).
3. submit the appropriate parameters to `bulk_raw_query` 

```
# before
aliased_query -> aliased_query_impl -> raw_query -> bulk_raw_query

# after
a. extract param generation from `alias_query` to `aliased_query_params`
b. extract param generation from `raw_query` to `raw_query_params`

                       ___raw_query_params(error)_
                      /                           \
aliased_query_params--                             ---> bulk_raw_query
                      \___raw_query_params(perf)__/

```


Fixes ISP-22

evanpurkhiser committed:
ref(js): Route name always after `path` (#39397)

evanpurkhiser committed:
ref(js): Fix missing types in note/mentionStyle (#39388)

evanpurkhiser committed:
ref(ts): Small style change in organizationDeveloperSettings (#39396)

evanpurkhiser committed:
ref(ui): Make mention suggestions look like our dropdowns (#39389)

Commit:
fix(metrics): Make calls to snql func more explicit (#39337)

Improves calls to snql attr of `SingularEntityDerivedMetric` by passing
explicit arg names of the snql arguments expected by these functions

matejminar committed:
feat(sampling): Add gatsby and svelte to supported sdks (#39400)

Commit:
fix(metrics): Fix flaky test [TET-423] (#39331)

kamilogorek committed:
ref(processor): Use symbolic-sourcemapcache for JavaScript Sourcemap processing (#38551)

This PR attempts to replace the currently used `rust-sourcemap` crate
and it's symbolic python bindings, with `symbolic-sourcemapcache` crate.

It makes the whole processing pipeline easier to maintain, as it pushes
some work directly to Symbolic, as well as we get better function names
due to better scope resolution and in some cases better file URLs.

Other than that, we don't use `SourceView` anymore, as it seemed like an
unnecessary layer of abstraction for something that is used only for
`context_lines` extraction. We cache `utf-8` decoded sources directly
now, as this way we can encode them only once for `SmCache` instance
initialization, and use the source directly otherwise for context lines
extraction.

Some tests had to updated to express current behavior.

The notable thing is `useless_fn_names = ["<anonymous>",
"__webpack_require__", "__webpack_modules__"]`, which is mostly for
`production` mode of webpack, that by default trims all the function
names, and we decided to fallback to the minified names in those cases
instead (this was already the old behavior).

It should be possible to extract something better, but we'd need to
parse all `sourceContents` from sourcemap to do that, as the only thing
we can get better function name for the case mentioned above, is if we
look at the right-hand side of default node export, in form of
`module.exports = function foo () {}`. This should give us `foo`, yet
the only thing we can extract is `module.exports`, as minified form of
this expression in webpack production mode is `module.exports = function
() {}`.

priscilawebdev committed:
ref(sampling): Update Performance Metrics Link - (#39401)

Commit:
feat(metrics): Add support for custom measurements [TET-228] (#39282)

edwardgou-sentry committed:
feat(dashboards): Adds search bar validation to release health widgets and cleans up some custom measurement config  (#39285)

Updates smartSearchBar to optionally highlight invalid tokens.
Also cleans up and fixes some typos for custom measurement config in
search parser.

edwardgou-sentry committed:
feat(dashboards): disables Widget Builder save button when query condition is invalid (#39350)

Disables widget builder save button if query condition is invalid

narsaynorath committed:
fix(discover): Filter for falsy is_homepage in discover endpoints (#39341)

A user shouldn't be able to interact with a homepage query outside of
the homepage endpoint. This adds filtering so we can differentiate the
two.

Zylphrex committed:
tests: disable flakey ingest consumer kafka test (#39404)

This tests failed in 2 runs in a row and is blocking deploys right now
https://github.com/getsentry/sentry/actions/runs/3143660577/jobs/5109409440

wmak committed:
fix(discover): Interval dropdown wrong side (#39362)

- Remove alignMenu right, which was causing the dropdown to render on
the right even when the screen width was very narrow


### Preview

#### Before

![image](https://user-images.githubusercontent.com/4205004/192623356-01e2071a-f0df-47f8-90e9-87586e3913dd.png)

#### After

![image](https://user-images.githubusercontent.com/4205004/192623393-3d96ffbe-f3cc-40bc-8c5c-a93686da9db8.png)

wmak committed:
fix(discover): Apply saved interval to query list (#39359)

- this uses the saved interval on the query list as well when available
- this removes the graph smoothing since there can be custom intervals
now

wmak committed:
chore(discover): Update the interval badge to new (#39406)

- This changes the badge from beta -> new
- Uses the space prop instead of a custom style

narsaynorath committed:
feat(discover): Open Discover with saved homepage query if it exists (#39368)

If you have a saved homepage query, render the landing page with that
query when clicking "Discover" in the sidebar. This should only fetch
the homepage query when navigating to the landing page by clicking the
sidebar (i.e. prebuilt queries and other saved queries will not fetch
this data)

k-fish committed:
feat(perf-issues): Change truncation check to be dependent on source (#39405)

### Summary
Source is likely safe to remove from the truncation check, since it's
not the actual problem itself (duplicated calls) and is instead related
to its actionability. We can check detection for just source
no-truncation before deciding if we need to remove the truncation check
on the offending spans as well.

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

Zylphrex committed:
fix(profiling): Docs link should go to profiling docs (#39411)

lynnagara committed:
test: Consolidate eventstream tests (#39366)

Move all the eventstream tests into one module. Also turn on type checking for test_synchronized.py.

kamilogorek committed:
ref(processor): Reduce number of feature flag queries for js processor (#39403)

Ask only once when initializing `JavaScriptStacktraceProcessor`, instead
of `1+2N` times (additional 2 for each frame).

scttcper committed:
feat(issues): Add back edit issue owners modal (#39385)

narsaynorath committed:
fix(discover): Check for discover-query before requesting homepage (#39419)

`discover-query` is the feature flag that allows users to save queries,
so the endpoint is also feature flagged with this. Don't fetch this
resource in the frontend if the user doesn't have `discover-query`.

barkbarkimashark committed:
fix(perf-issue): fix merging results with None data returned (#39421)

Missed a scenario when the bulk results returns None data causing some
merging issues.

Fixes SENTRY-W3V

vuluongj20 committed:
ref(pageFilters): Update desync alert (#39390)

A few improvements to the desynced filter alert:
* An updated default message that mentions which filters are desynced
* Ability to add custom, page-specific desync messages. These custom
messages are useful for pages where there might be a different reason
why the filters are desynced, including Dashboards and, at some point in
the future, Discover.

**Before:**
<img width="1199" alt="Screen Shot 2022-09-27 at 5 19 03 PM"
src="https://user-images.githubusercontent.com/44172267/192659790-111c1b43-fa80-4429-8358-eb291f33a78b.png">

**After, default message:**
<img width="1199" alt="Screen Shot 2022-09-27 at 5 13 37 PM"
src="https://user-images.githubusercontent.com/44172267/192659701-9453c8d2-f209-4708-b456-920f43925515.png">

**After, custom message in Dashboards:**
<img width="1199" alt="Screen Shot 2022-09-27 at 5 13 45 PM"
src="https://user-images.githubusercontent.com/44172267/192659706-68dd5038-6d65-4abf-aaeb-dd29853fbf38.png">

dashed committed:
chore(hybrid-cloud): Delegate pipeline redis store to propagate customer domain (#39016)

k-fish committed:
feat(perf-issues): Add new option and graceful rollout for ext detector (#39407)

This adds a new option for the extended n+1 detector, on top of adding
code to make rollout of extended detectors more graceful by removing
duplicate problems before they are emitted to our grouping logic. This
should make it safe to leave both detector rates (original and extended)
at 1.0 while rolling it out without any extra code running.

Commit:
fix(commit-context): Filter db query for not null (#39371)

Fixes
[SENTRY-W3A](https://sentry.io/organizations/sentry/issues/3615924632/?referrer=jira_integration)

wedamija committed:
fix(tests): Fix another flakey test caused by using `RedisSnubaTSDB` (#39425)

Need to freeze the hour on this test as well

ceorourke committed:
ref(perf issues): Enable ignore in a time period (#39120)

Enable ignoring a performance issue in a time period e.g. ignore this
until it happens 10x / hr or ignore until 10 users experience it in an
hour.

edwardgou-sentry committed:
fix(discover): fixes bug where previous series would not format tooltip value (#39417)

Fixes an issue where previous series in discover would not format the tooltip value correctly

edwardgou-sentry committed:
feat(dashboards): Prevent widget builder preview from firing request if widget query condition is invalid (#39414)

Updates the widget builder preview to not send widget query requests if
the widget query condition is invalid.
This is to mitigate the amount of invalid requests we send to backend
apis since we already know that they will fail.

k-fish committed:
feat(perf-issues): Add option to ux for n+1 ext  (#39428)

### Summary
This adds the option to rollout the extended n+1 detector.

corps committed:
An accidental edit in response to code review, resume random group id… (#39429)

https://github.com/getsentry/sentry/pull/38988#discussion_r982725084

wmak committed:
fix(discover): Don't rerender table when chart props change (#39418)

- This fixes a bug where changing an attribute of a chart (eg. display)
the table would rerender & rerequest data
- This basically saves the previously loaded view into the table state,
so even when the confirmedQuery value flips we don't reload the table
unless the previously loaded view is different
- This is because whenever a value of the EventView changes the
confirmedQuery value toggles to true then false again to cause a render
- This has to happen because the confirmedQuery has to stay in false
since statsPeriod/project is set before we can change its value, which
means the confirmation on >10 projects will break

beezz committed:
feat(buffer): Allow override of process function of redis buffer impl (#39399)

This change allows implementations based on the `RedisBuffer` class to
override the actually processing functionality. More info
https://github.com/getsentry/getsentry/pull/8421

Commit:
fix(monitors): Correct chart series timestamp values (#39427)

Don't need to `.toString` the timestamp values, in fact the chart
doesn't seem to work if they are `.toString`'ed

Before:
<img width="1188" alt="image"
src="https://user-images.githubusercontent.com/9372512/192858985-02b8ef88-d7c8-4b74-a31e-43ea0aa3a21a.png">


After:
<img width="1188" alt="image"
src="https://user-images.githubusercontent.com/9372512/192858937-9dfeccca-e246-454f-9ae8-b5e7569ca2f1.png">

Commit:
feat(issue-alerts): add event_id to RuleFireHistory (#39363)

Records `event_id` in `RuleFireHistory` to query last triggered event
for an issue alert.
[jira](https://getsentry.atlassian.net/browse/FEEDBACK-806)
[Frontend pr](https://github.com/getsentry/sentry/pull/39364)

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

Commit:
feat(issue-alert): link to last triggered event instead of issue (#39364)

If backend provides the last triggered `eventId` of an alert issue, link
to event page instead of issue page.
[jira](https://getsentry.atlassian.net/browse/FEEDBACK-806)
[backend pr](https://github.com/getsentry/sentry/pull/39363)

wedamija committed:
tests: Include testutils in test paths (#39386)

This allows asserts in testutils to work as expected

k-fish committed:
ref(perf-issues): Add platform to group metric (#39436)

### Summary
We're using `event_manager.create_performance_group_transaction.count`
to check created perf-issue groups at the moment, we could alternatively
add a bool to `group.created`, but either way I think we should have a
breakdown of create perf issue groups by platform.

malwilley committed:
ref(ui): Extract STACKTRACE_PREVIEW_TOOLTIP_DELAY to a global constant (#39426)

evanpurkhiser committed:
ref(js): Remove `RouteWithName` in a number of places (#39398)

These don't need to be Routes with name

evanpurkhiser committed:
ref(js): Replace MonitorModel w/ transformData opt (#39383)

nhsiehgit committed:
feat: add optional ref to useKeyPress (#39370)

<!-- Describe your PR here. -->

Adds optional ref object to the useKeyPress util method (as opposed to
just window)

This allows us to attach keypress events to multiple components within
the same window.

Mainly borrowed from
[here](https://newdevzone.com/posts/react-how-to-navigate-through-list-by-arrow-keys)

<!--

  Sentry employees and contractors can delete or ignore the following.

-->

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated
in the State of Delaware in 2015 as Functional Software, Inc. and is
gonna need some rights from me in order to utilize my contributions in
this here PR. So here's the deal: I retain all rights, title and
interest in and to my contributions, and by keeping this boilerplate
intact I confirm that Sentry can use, modify, copy, and redistribute my
contributions, under Sentry's choice of terms.

shruthilayaj committed:
feat(discover): Track how people interact with baseline (#39413)

Adding analytics to the metrics baseline.

AniketDas-Tekky committed:
feat(api-gateway): Unit test helpers (#39424)

These functions will help with the creation of new test cases for the
API gateway

dashed committed:
chore(hybrid-cloud): Turn off customer domain middleware by default (#39433)

wedamija committed:
fix(search): Fix referrer in issue search (#39443)

This is messing up some stats in snuba to make it look like we're making
a lot more queries.

malwilley committed:
feat(perf-issues): Re-enable ignore options by time window for all issue types [UP-172] (#39441)

Performance issues supports these options now
(https://github.com/getsentry/sentry/pull/39120)

So this reverts the work done in these PRs:

https://github.com/getsentry/sentry/pull/38756 

https://github.com/getsentry/sentry/pull/38540

dashed committed:
fix(oauth2): Do not generate login redirect URL too early (#39449)

victorelec14 committed:
fix(issues): getsentry/self-hosted#1672 next button redirection (#38377)

The next button redirects you to a url that is wrong because a slash was
missing in the url.

In the following lines of code it looks like it is well written.

Thanks

Related Issue: https://github.com/getsentry/self-hosted/issues/1672




<!-- Describe your PR here. -->



<!--

  Sentry employees and contractors can delete or ignore the following.

-->

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated
in the State of Delaware in 2015 as Functional Software, Inc. and is
gonna need some rights from me in order to utilize my contributions in
this here PR. So here's the deal: I retain all rights, title and
interest in and to my contributions, and by keeping this boilerplate
intact I confirm that Sentry can use, modify, copy, and redistribute my
contributions, under Sentry's choice of terms.

Co-authored-by: Armen Zambrano G <armenzg@sentry.io>

priscilawebdev committed:
feat(dynamic-sampling): Add dynamic sampling feature plans and upsell (#39455)

narsaynorath committed:
feat(discover): Add save actions to update homepage query (#39431)

Adds a button and context menu option (in the manage page) to save the
current query as the Discover homepage

Commit:
feat(metrics): Add support for 'tuple' conditions [TET-319] (#39451)

k-fish committed:
fix(perf-issues): Fix bug with span evidence and parent span (#39463)

When the parent span is the root span for the transaction, it's not in
the spans list, and causes an error since the span doesn't appear.

Fixes JAVASCRIPT-2ANQ

robinrendle committed:
fix(copy): modal header and events tab (#39444)

1. Changing `Events` tab to `All Events` on the Issue Details page for
consistency
2. Updated the Ownership modal with the correct heading size

rjo100 committed:
ref(perf_issues): Initial post_process_group refactor (ISP-11) (#39369)

Initial refactor of `post_process_group`. Will refactor `group_inbox`
stuff in a subsequent PR. Then, once we have confidence in the changes,
will upgrade to a more robust pipeline system

narsaynorath committed:
feat(discover): Add alpha badge to Use as Discover Home (#39462)

k-fish committed:
fix(perf-issues): Need to splat array in span evidence (#39465)

Array was being pushed into directly on the event, we want to make a new
array reference so we don't mutate the existing ref.

Abdkhan14 committed:
fix(discover): Added check to render prepend columns along with the o… (#39416)

Added check to render prepend columns along with the other header
columns. This gets rid of the quick flicker when we click into saved
queries in discover.

Before: 
<img width="1004" alt="Screen Shot 2022-09-28 at 1 51 07 PM"
src="https://user-images.githubusercontent.com/60121741/192853628-6c446328-18f6-4c1c-a90c-a90b4ca54132.png">

<img width="1253" alt="Screen Shot 2022-09-28 at 1 51 27 PM"
src="https://user-images.githubusercontent.com/60121741/192853694-71751f6d-4a19-4422-aee4-db287673f634.png">

After:
<img width="1004" alt="Screen Shot 2022-09-28 at 1 49 14 PM"
src="https://user-images.githubusercontent.com/60121741/192853253-ace8ce36-8e5b-40f4-9f72-d38b99220680.png">

<img width="1256" alt="Screen Shot 2022-09-28 at 1 49 46 PM"
src="https://user-images.githubusercontent.com/60121741/192853343-3d1526af-7754-462f-b4b9-b80a7f511be3.png">

Co-authored-by: Abdullah Khan <abdullahkhan@P40P69L0VQ-Abdullah-Khan.local>

Commit:
feat(issues): adds more fields to issue_details.viewed event (#39439)

This PR adds a whole set of new fields to the `issue_details.viewed`
event so we can start tracking the feature penetration of things like
releases, source maps, etc.

Commit:
ref(issue): Remove unused IssueList (#39324)

Removes unused issue list (depends on
https://github.com/getsentry/sentry/pull/39323)

Commit:
fix(commit-context): Update GitHub fetch blame query (#39438)

udameli committed:
test(perf-issues): Improve how performance issues are created in tests (#39293)

This PR updates `Factories.store_event` and `load_data` to support
creation of performance groups for transactions. Once this PR is merged,
I will update all instances of `hack_pull_out_data`.

Resolved ISP-16

DominikB2014 committed:
feat(issues) Implement new all events tab for errors (#39466)

This brings the new all events tab for errors under a feature flag, we
still have to add attachments as part of the table before we fully
release it.

<img width="999" alt="image"
src="https://user-images.githubusercontent.com/44422760/193083213-c316f88e-15ea-4e20-aa83-5f29982b6ec3.png">

scttcper committed:
feat(ecosystem): Track stacktrace link results (#39448)

Commit:
feat(migration): Add suspect_committer_auto_assignment column (#39382)

## Objective:
Adds the new setting to auto assign Issues to the Suspect Committer.

k-fish committed:
fix(trends): Keep period consistent (#39437)

When on the default period for landing, clicking into trends doesn't
keep the default period, which causes odd results in the trends view.
Whether we should link directly to transaction summary is a separate
discussion, for now we can just fix this part.

nhsiehgit committed:
fix: capture current ref outseide of useEffect (#39477)

<!-- Describe your PR here. -->
updateUseKeyPress to capture current ref in case ref target has changed


<!--

  Sentry employees and contractors can delete or ignore the following.

-->

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated
in the State of Delaware in 2015 as Functional Software, Inc. and is
gonna need some rights from me in order to utilize my contributions in
this here PR. So here's the deal: I retain all rights, title and
interest in and to my contributions, and by keeping this boilerplate
intact I confirm that Sentry can use, modify, copy, and redistribute my
contributions, under Sentry's choice of terms.

0Calories committed:
fix(perf-issues): Make resources section mobile responsive + contextual links (#39430)

Commit:
feat(profiling): add discord link to feedback modal (#39461)

This PR adds the ability to render a `secondaryAction` in the footer of
the `FeatureFeedback` modal. In our case we'd like to link out directly
to the `#profiling` discord.


![image](https://user-images.githubusercontent.com/7349258/193055617-44d9784c-e250-4605-92b3-a6e57a1cdc7e.png)


![image](https://user-images.githubusercontent.com/7349258/193057291-ef872f6d-1257-41f7-aca5-6fe0fc7ab4c5.png)

scttcper committed:
test(ui): Convert stacktracelink test to RTL (#39476)

gggritso committed:
fix(perf-issues): Normalization more database spans hashes (#39473)

While checking performance issues in Laravel projects, we noticed that
queries `SELECT ... IN (?, ?)` and `SELECT ... IN (?, ?, ?)` had
different hashes, which they're not supposed to. The cause is that `IN`
normalization only applies to `db` spans, but queries can also have the
ops `db.query` and `db.sql.query`. Expands the normalization to also
include those spans.

I opted to let `span_op` accept a list in lieu of accepting a regex or a
prefix, because I thought it was the most explicit (i.e., regexes are
easy to get wrong, and some ops like `db.connection` and `db.redis`
probably don't need this normalization, so prefix is too loose) but you
tell me!

This will improve span grouping and performance issue detection, since
we rely on the hashes generated from the span grouping code.

cmanallen committed:
feat(replays): Add delete method for the replay details resource (#39078)

Deletes a replay's recording segments. A replay has a one to many
relationships with its segments. The segment data is stored in blob
storage and has to be removed. Due to the I/O requirements we perform
the operation asynchronously to provide fast API responses and to give
ourselves good retry semantics.

closes https://github.com/getsentry/replay-backend/issues/136

depends https://github.com/getsentry/snuba/pull/3180
depends https://github.com/getsentry/snuba/pull/3181

Co-authored-by: Josh Ferge <josh.ferge@sentry.io>

barkbarkimashark committed:
fix(perf-issue): revert parallel snuba search  (#39479)

We're observing some extra search queries in datadog after introducing
search query parallelism for multiple GroupCategories. Want to roll the
change back to see if the parallelism is causing the extra queries or
whether its from a different source.

Reverted the following commits:
* 51c2e3c836dc2a9345046f37642bf14565730019
* 2beedf8e73e4f6e51a1cdd34d22fe04f4ef231c5
* d868287f735d862dac94303587daa9f0666f37ba

evanpurkhiser committed:
feat(perf-issues): Reenable share (#39478)

evanpurkhiser committed:
ref(js): Remove UNSAFE_componentWillUpdate from organizationAuth (#39221)

ceorourke committed:
ref(tests): Replace usages of mock django timezone with freezegun (#39321)

Use freezegun everywhere for consistency 🧊 🔫 we use this in many more
places than `@mock.patch("django.utils.timezone.now")` already so this
is just replacing the remaining usages so every test is using the same
thing and it's not confusing when you see both and wonder which one to
use.

Commit:
ref(monitors): Convert MonitorStats to FC (#39447)

maxiuyuan committed:
conf(Snowflake): Enable snowflake ID without using feature flag (#39474)

evanpurkhiser committed:
ref(js): Convert note/input to a FC (#39391)

nhsiehgit committed:
useKeyPress takes an html element rather than ref (#39482)

<!-- Describe your PR here. -->

rather than passing a ref - lets just pass an html element and rely on
the implementation to keep track of dom mounting changes

RE: https://github.com/getsentry/sentry/pull/39370#discussion_r983923968

<!--

  Sentry employees and contractors can delete or ignore the following.

-->

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated
in the State of Delaware in 2015 as Functional Software, Inc. and is
gonna need some rights from me in order to utilize my contributions in
this here PR. So here's the deal: I retain all rights, title and
interest in and to my contributions, and by keeping this boilerplate
intact I confirm that Sentry can use, modify, copy, and redistribute my
contributions, under Sentry's choice of terms.

ryan953 committed:
ref(replay): Divide replay spans into memory and network arrays once, inside ReplayReader (#39354)

Refactor to remove `isMemorySpan` and `isNetworkSpan` from the
public interface if ReplayReader, and just export the lists that
components need.

This means components don't need to filter things themselves, so object
(array) references will be more stable.

scttcper committed:
feat(workflow): Revert default project ownership fallthrough (#39486)

ryan953 committed:
fix(replays): Use the updated Replay SDK blockClass to prevent replays from capturing replays (#39487)

The up-to-date default classNames are listed here: https://github.com/getsentry/sentry-replay

danecando committed:
feat(replays): Remove replay-metadata-table on Issues Details  (#39420)

### Changes
- Adds support for passing additional buttons to the ReplayController
- Updates eventReplay content to the new design featuring the view full
replay and replays alpha badge
- Removes event replay table content
- Removes test assertions on removed replay table content

### Notes
Will remove the EventDataSection wrapper around the replay content in
the PR where it moves into the breadcrumbs section

### Checklist
- Tested that the new "View Full Replay" button works correctly and
links to the replay detail page
- Tested that new ReplayView and ReplayController props doesn't cause
any issues when they are defined or undefined. There is no changes to
the original controls

Initial PR for #39026 

![Screen Shot 2022-09-28 at 12 57 45
PM](https://user-images.githubusercontent.com/3721977/192844687-ceb4bcbe-5112-4353-934b-8e27f600e463.png)


<!--

  Sentry employees and contractors can delete or ignore the following.

-->

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated
in the State of Delaware in 2015 as Functional Software, Inc. and is
gonna need some rights from me in order to utilize my contributions in
this here PR. So here's the deal: I retain all rights, title and
interest in and to my contributions, and by keeping this boilerplate
intact I confirm that Sentry can use, modify, copy, and redistribute my
contributions, under Sentry's choice of terms.

danecando committed:
fix(replays): adds ReplayReaderParams test stub (#39415)

### Changes
- Adds a ReplayReaderParams overridable test stub
- Updates `replayContent.spec.tsx` to use the new stub

Returns mock data that can be used to instantiate a ReplayReader for
testing things that rely on a replay

jesus4497 committed:
feat(replays): Add informative tooltip to activity column (#39471)

### Changes
- Added tooltip icon for the activity column to describe a bit more its
meaning.

### Checklist
- [x] Made sure it was working as expected in Performance, Issues and
Replays pages
- [x] Tested different resolutions and theme colors

![image](https://user-images.githubusercontent.com/39612839/193148460-f320369f-c9ca-452a-8a00-967a40795ec6.png)

![image](https://user-images.githubusercontent.com/39612839/193148552-3da18bc4-3c98-43db-a2cd-db2bbb689619.png)


Closes #38478

evanpurkhiser committed:
ref(js): Rename note modelId to noteId (#39492)

evanpurkhiser committed:
ref(css): Remove extraneous selector in note (#39496)

evanpurkhiser committed:
ref(js): Better comments on note/input props (#39494)

dashed committed:
chore(hybrid-cloud): Move DedupeCookiesMiddleware to getsentry (#39442)

priscilawebdev committed:
ref(feature-flag): Add feature flag for a DY CTA in the performance page - (#39507)

Zylphrex committed:
fix(tests): More django cache clears in ingest consumer tests (#39481)

priscilawebdev committed:
feat(sampling) Add feature flag for DS CTA in performance (follow-up) (#39509)

barkbarkimashark committed:
fix(perf-issue): re-revert parallel snuba search (#39490)

Undoes this revert https://github.com/getsentry/sentry/pull/39479

Looks like this code is likely not the reason for the increase in search
queries - bringing it back.

rjo100 committed:
ref(perf-issues): Move post_process_group inbox logic (ISP-11) (#39480)

Final part of splitting `post_process_group` into its own self contained
functions.

evanpurkhiser committed:
fix(js): Add @types/prismjs (#39497)

scttcper committed:
ref(ui): Convert stacktracelink modal to FC (#39500)

barkbarkimashark committed:
feat(perf-issue): introduce multiple groups to post process forwarder (#39393)

With the introduction of Performance issues, the assumption that an
event will create at most one Group must change. Errors will always
correspond to a single Group, but transactions can support up to 10
different types of Groups being created to accommodate for the different
detection methods for a single transaction.

To support `post_process_group` working for transactions that result in
multiple Groups being created, we need to modify the
`EventStream.insert` and `PostProcessForwarderWorker` to accept a
datastructure like:
```
{
  1: {
    "is_new": False,
    "is_new_group_environment": False,
    "is_regression": False,
  },
  2: {
    "is_new": True,
    "is_new_group_environment": False,
    "is_regression": False,
  }
}
```
that describes the groups that were created and its associated task
state.


Fixes ISP-13

danecando committed:
feat(replays): Event detail Replay navigation button (#39512)

Adds a Replay button that links to the `#breadcrumbs` anchor if there is a replayId related to the event.

![Screen Shot 2022-09-27 at 11 43 08
AM](https://user-images.githubusercontent.com/3721977/192572692-5d41c435-a94f-4d4b-9362-9d522a6bc3a3.png)

Closes #38956

evanpurkhiser committed:
ref(js): Remove old Global Selection Header components (#39499)

These aren't used anymore :tada:

evanpurkhiser committed:
ref(js): Better type for note Mention (#39495)

barkbarkimashark committed:
fix(perf-issue): revert handling multiple groups in post process forwarder (#39519)

Revert of https://github.com/getsentry/sentry/pull/39393

jesus4497 committed:
feat(replays): Unit tests to Issues Replays List page (#39374)

Added unit tests for the `groupReplays.tsx`. 

Found and fixed an issue where `fetchError` wasn't being passed into `ReplaysTable` component, it is now.

Closes #38842

dashed committed:
fix(gpu-context): Properly insert known data types for gpu context (#39506)

Commit:
feat(metrics): Add `team_key_transaction` to metrics layer [TET-325 TET-426] (#39412)

Adds support for `team_key_transaction` to be added
to the metrics layer. As of these changes, we should be able to
select, groupBy and orderBy team_key_transaction

This PR achieves the following by
- Adds a new derived operation `team_key_transaction` and accepts
transaction.duration MRI as a metric mri
- Modifying `MetricsGroupByField` to also accept an instance of
`MetricField`
- Modifying `MetricsExpressionBase` that has a
`generate_groupby_statement` that generates SNQL but also validates that
the metrics expression supports grouping by
- Refactor shared logic out of `SnubaQueryBuilder._build_groupby` and
`_get_group_limit_filters` into a static method
`SnubaQueryBuilder.generate_snql_for_groupby_field`

Commit:
feat(metrics): Allow `team_key_transaction` to be used as a condition [TET-427 TET-325] (#39535)

Adds capability to use `team_key_transaction` as a condition in the
query. It achieves this by introducing `ConditionMetricField` which in
the future should replace the `MetricQuery` accepting snuba conditions

Commit:
feat(profiling): switch give feedback for join discord btn (#39542)

This PR swaps out the "Give Feedback" button with a "Join Discord"
button in an effort to drive more traffic into discord for faster direct
feedback.

dashed committed:
fix: Remove unncessary truthy check for AsyncComponent (#39537)

adinauer committed:
Add Java SDK to supported SDKs for DS (#39503)

dashed committed:
fix(profiling): Remove useless expression (#39536)

sl0thentr0py committed:
feat(sampling): Add ruby and rails to supported DS SDKs list (#39544)

supported from [5.5.0](https://github.com/getsentry/sentry-ruby/releases/tag/5.5.0)

shruthilayaj committed:
fix(discover): Pagination on discover table (#39549)

Pagination wasn't working because we weren't handling the state 
of confirmedQuery properly. Handle pagination as if state
confirmedQuery is true.

danecando committed:
feat(replays): Move EventReplay into breadcrumbs section on Issue Details page (#39511)

Moves EventReplay into Breadcrumbs event entry.
- RRWebIntegration now stands alone at the bottom of eventEntries
- Adds test case for rendering EventReplay as part of Breadcrumbs

Also fixes a bug with ReplayReaderParams stub

### Checklist
- [x] Breadcrumb has no changes from previous style and behavior when
replayId is undefined
- [x] Checked to make sure no issues were introduced to the share view
- [x] Additional props introduced to breadcrumbs do not cause
incompatibilities elsewhere (breadcrumbs is only referenced in
eventEntry and tests)



`EventReplay` load error state (there's still some replays that have
associated replayId's that do not exist)

![Screen Shot 2022-09-26 at 3 37 25
PM](https://user-images.githubusercontent.com/3721977/192365396-e60c61e9-5d9b-4451-be9d-171e100ed661.png)

Replay rendered as part of the breadcrumbs event entry

![Screen Shot 2022-09-26 at 3 37 56
PM](https://user-images.githubusercontent.com/3721977/192365612-c9bd5b0b-8708-4c17-a9ec-34d3510a18ff.png)

Event rendered with no associated replayId

![Screen Shot 2022-09-26 at 3 41 21
PM](https://user-images.githubusercontent.com/3721977/192365733-371d23ef-98bc-4d57-ae22-a146118afb04.png)


Closes #39026

ryan953 committed:
feat(replays): Fetch error title when querying for issues related to a replay (#39488)

Fetch the error title because captured replay error breadcrumbs might
not have the message field set, and we'll need something to show

With the title field added (and group.id wasn't ever returning anything
-> removed now):
![Screen Shot 2022-09-29 at 3 15 54
PM](https://user-images.githubusercontent.com/187460/193152418-322c35f6-481b-4a50-adc2-2af243c13b6c.png)

Required by #39489
See #39339

evanpurkhiser committed:
fix(ui): Remove overflow scroll on diff modal (it already has it) (#39555)

evanpurkhiser committed:
ref(js): Remove UNSAFE_componentWillReceiveProps from tagFilter (#39220)

oioki committed:
Add `replays.delete_replay` queue (#39560)

Before switching the task
(https://github.com/getsentry/sentry/pull/39520), we need to create the
queue.

malwilley committed:
fix(project-selector): Fix 'Select All Projects' behavior [UP-173] (#39515)

If you select a project, then click "Select All Projects",
`actions.close()` triggers the `onClose` handler which has outdated
state, reverting the component back to the old selection.

malwilley committed:
ref(ui): Make stack trace tooltip generic for more issue types (#39483)

Moves `StackTraceTooltip` into a parent `GroupPreviewTooltip` component
to allow us to add previews for other issue types.

evanpurkhiser committed:
ref(ui): Don't export SettingsIconLink from HeaderItem (#39565)

cmanallen committed:
feat(Replays): Make replays kafka publisher a global (#39564)

closes https://github.com/getsentry/replay-backend/issues/145

edwardgou-sentry committed:
fix(discover): Displays custom units in custom perf metrics sidebar in event detail pagge (#39543)

When a custom performance metric has a custom unit supplied, render that
unit in the custom performance metrics side bar in the event details
pages

Commit:
fix(commit-context): Frames start from the bottom of the error stack (#39533)

## Objective:
Frames start from the bottom of the error stack. So currently the Commit
Context is matching on frames of base libraries. Whereas the value of
Commit Context is in the top-level frame that cause the exception.

k-fish committed:
ref(perf): Add full-projects function to EventView (#39563)

We often try to convert the eventview list of project ids (the
confusingly named 'project') into full project objects. This function
should let us remove that boilerplate code in a bunch of places moving
forward.

ryan953 committed:
fix(replay): Use the issue title when rendering Error breadcrumbs inside Replay Details (#39489)

Depends on #39488

Fixes #39339

dashed committed:
fix(mep): Fix unreachable code in test (#39538)

JoshFerge committed:
chore(replays): put replay deletions on their own celery queue (#39520)

for better isolation, let's put our replay deletion tasks on their own
queue. depends on https://github.com/getsentry/ops/pull/5391

scttcper committed:
fix(issues): Remove padding from checkmark icon (#39574)

JoshFerge committed:
chore(replays): add replay deletion to sampled tasks (#39525)

sample APM data for the replay deletion celery task (this is low
throughput so default should be fine).

nikhars committed:
feat(indexer): Allow mechanism to skip indexing tag values (#39139)

The original PR https://github.com/getsentry/sentry/pull/38837 which got
merged mistakenly had the code to treat the rollout option as
percentage. It was decided that it is not necessary for it to be
percentage based. This PR is the same as the above linked PR but with
the option value treated as a boolean since that is what is setup in
production environment.
